### PR TITLE
Up-hint HUD, ported from Aurora + Look Up/Down refactor

### LINKED ISSUE #60: Look Up verb issues
#### Description of issue

Look up verb which should allow to see what above if there open space have several issues:
1. Vision doesn't reset if you walk out from open-space range
2. You can click Look Up verb while walking on stairs and move your camera over two levels, even if there no open-space above.

#### Difference between expected and actual behavior
Look Up verb should reset after mob get out from open-space range or if something prevent see above (vision block, moving from one level to another, etc)

#### Steps to reproduce
1. open-space above + Look Up verb + Movement

Result: Weird effect like eye stuck from above

2. open-space above + Ladder / stairs which have connected ladder / stairs above + Look Up verb 

Result: Total blackness or moving client eye to another level aka bluespace multiz x-ray vision :D

#### Length of time in which bug has been known to occur

This bug exist on any latest baystation and nebula versions since z-mimic appearance

#### Client version, Server revision & Game ID

Client: 512 (stable)
Server: 512 (stable)

#### Issue bingo

- [X] Issue could be reproduced at least once
- [X] Issue could be reproduced by different players
- [X] Issue could be reproduced in multiple rounds
- [X] Issue happened in a recent (less than 7 days ago) round
- [X] [Couldn't find an existing issue about this](https://github.com/NebulaSS13/Nebula/issues)

### PR DESCRIPTION
**Pull request by ghostsheet**
As suggested, up hint hud was ported from aurora
It calls lookup() when clicked upon
It comes in the 6 different Ui style (I made the minimalist myself, v proud of my coder sprite tyvm)
The up hint HUD works for borgs and human only (while inside mechs as well)

Refactored Look up and added a look down verb.
Fixes #60

klix4Ka62K

### CODE REVIEW

File: code/modules/mob/living/silicon/robot/robot_movement.dm [hunk 1]
```diff
@@ -16,6 +16,15 @@
 	. = ..()
 
 
+/mob/living/silicon/robot/Move()
+	.=..()
```
Comment (deathride58) [inline code/modules/mob/living/silicon/robot/robot_movement.dm]: This is a tiny nitpick but it's also one of my bigger pet peeves, grmbl grmbl
```suggestion
	. = ..()
```

**Review by deathride58 — COMMENTED** [review]
I believe @Lohikar may be interested in this here.

Aside from that, there's just one tiny thing I noticed.

File: code/modules/mechs/mech_movement.dm [hunk 2]
```diff
@@ -6,8 +6,20 @@
...
-	if(. && !istype(loc, /turf/space))
-		playsound(src.loc, mech_step_sound, 40, 1)
+	if(.)
+		if(!istype(loc, /turf/space))
+			playsound(src.loc, mech_step_sound, 40, 1)
+
+		var/turf/T = src.loc
+		if (!isturf(T))	return
+
+		var/turf/B = GetAbove(T)
```
Comment (MistakeNot4892) [inline code/modules/mechs/mech_movement.dm]: You don't seem to check `HasAbove()` anywhere in this block.

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 3]
```diff
@@ -121,6 +121,13 @@
 	if(.) //We moved
 		handle_exertion()
 		handle_leg_damage()
+	
+		if(client)
+			var/turf/T = src.loc
+			if (!isturf(T))	return
+
+			var/turf/B = GetAbove(T)
```
Comment (MistakeNot4892) [inline code/modules/mob/living/carbon/human/human_movement.dm]: Likewise, not checking `HasAbove()` and not checking if `GetAbove()` returned a usable value.

File: code/modules/mob/living/living.dm [hunk 4]
```diff
@@ -859,3 +859,24 @@ default behaviour is:
 
 /mob/living/proc/is_telekinetic()
 	return FALSE
+
+/mob/living/verb/lookup()
+	set name = "Look up"
```
Comment (MistakeNot4892) [inline code/modules/mob/living/living.dm]: Minor quibble, `Look Up` would fit with other verbs.

**Comment by MistakeNot4892** [comment]
Would it be feasible to have a Look Down verb using the hooks you're adding?

**Comment by MistakeNot4892** [comment]
Misclick, whoops.

File: code/modules/mechs/mech_movement.dm (hunk 2 above)
Comment (Lohikar) [reply code/modules/mechs/mech_movement.dm]: `GetAbove()` implies `HasAbove()`, checking both is redundant.

**Comment by ghostsheet** [comment]
I can't add a look down verb, not with the way the hacky way it's using boundoverlay from z mimic. It'd need to be it's own thing.

File: code/modules/client/ui_style.dm [hunk 5]
```diff
@@ -3,10 +3,10 @@
 /var/all_ui_styles = list(
 	"Midnight"     = 'icons/mob/screen/midnight.dmi',
 	"Orange"       = 'icons/mob/screen/orange.dmi',
-	"old"          = 'icons/mob/screen/old.dmi',
+	"Oold"          = 'icons/mob/screen/old.dmi',
```
Comment (quardbreak) [inline code/modules/client/ui_style.dm]: `Old`

**Comment by ghostsheet** [comment]
Added that look down verb, as loaf suggested

**Review by comma — APPROVED** [review]

**Pull request merged**